### PR TITLE
Fix some NftAllowanceRepository corner cases

### LINKED ISSUE #8354: RestJava NftAllowanceRepository returns incorrect allowances
### Description

NftAllowanceRepository returns incorrect allowances with certain spender / owner bound and token bound. Detail TBA

### Steps to reproduce

N/A

### Additional context

_No response_

### Hedera network

other

### Version

V0.106.0-SNAPSHOT

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes some NftAllowanceRepository corner cases

- Fix query with `token.id=a`
- Fix query with `token.id=gte:a&token.id=lte:a` and valid `account.id` range
- Fix query with `account.id=gt:a1&account.id=lte:a2&token.id=lt:t1`
- Fix multiple flaky tests in `AllowancesControllerTest`
- Add test configuration to turn on jooq SQL logging

**Related issue(s)**:

Fixes #8354 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerTest.java [hunk 1]
```diff
@@ -137,13 +137,12 @@ void nftAllowancesAlias() {
         // When
         var result = restClient
                 .get()
-                .uri("", new Base32().encodeAsString(entity.getAlias()))
+                .uri("", BaseEncoding.base32().omitPadding().encode(entity.getAlias()))
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerTest.java:140]: flaky because apache common base32 encoding adds padding `=` while HIP mandates no padding.

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerTest.java [hunk 2]
```diff
@@ -179,8 +178,11 @@ void nftAllowancesOrderAsc() {
                 .nftAllowance()
                 .customize(nfta -> nfta.owner(allowance1.getOwner()).approvedForAll(true))
                 .persist();
-        var uriParams =
-                "?account.id=gte:0.0.1000&account.id=lt:0.0.1002&owner=true&token.id=gt:0.0.1000&limit=1&order=asc";
+        var uriParams = "?account.id=gte:%s&account.id=lt:%s&owner=true&token.id=gt:%s&limit=1&order=asc"
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerTest.java:181]: any tests with hardcoded entity ids can be flaky.

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 3]
```diff
@@ -188,258 +184,297 @@ private void setupNftAllowances() {
...
+                                        List.of(fromIndex(
+                                                LTE, tokenIds,
+                                                0)), // new EntityIdRangeParameter(LTE, EntityId.of(tokenIds.get(0)))),
+                                        false,
+                                        ParameterNames.TOKEN_ID))
+                                .limit(6)
+                                .order(Direction.DESC)
+                                .build(),
+                        List.of(new Tuple(1, 1, 2), new Tuple(1, 1, 0)),
+                        "given owner 1, limit 4, and desc, expect (2, 0) >= (1, 2), (1, 0) > (0, *)"),
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:331]: the original test spec expects (2, 1), (1, 2), (1, 0), and (0, 1). It's wrong because (2, 1) > (2, 0), and (0, 1) < (0, 2).

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 4]
```diff
@@ -188,258 +184,297 @@ private void setupNftAllowances() {
...
+                                        ParameterNames.ACCOUNT_ID))
+                                .tokenIds(new Bound(
+                                        List.of(fromIndex(GT, tokenIds, 0), fromIndex(LT, tokenIds, 2)),
+                                        false,
+                                        ParameterNames.TOKEN_ID))
+                                .limit(4)
+                                .order(Direction.ASC)
+                                .build(),
+                        List.of(new Tuple(0, 0, 2), new Tuple(0, 1, 1), new Tuple(0, 2, 0)),
+                        "given owner 0, limit 4, and asc, expect (0, 0) < (0, 2), (1, 1), (2, 0) < (2, 2)"));
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:454]: two new test specs

File: hedera-mirror-rest-java/src/test/resources/config/application.yml [hunk 5]
```diff
@@ -1,3 +1,10 @@
+#logging:
+#  level:
+#    org:
+#      jooq:
+#        tools:
+#          LoggerListener: DEBUG # logs SQL generated by jooq, uncomment it on for debugging purposes
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/resources/config/application.yml:6]: it's commented out by default, handy when we need to check the generated SQL to debug

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 6]
```diff
@@ -188,258 +184,293 @@ private void setupNftAllowances() {
...
-                    .containsExactlyElementsOf(expectedNftAllowances);
+        var softAssertion = new SoftAssertions();
+        for (int i = 0; i < testSpecs.size(); i++) {
+            var testSpec = testSpecs.get(i);
+            var request = testSpec.request();
+            var expected = testSpec.expected().stream().map(nftAllowances::get).toList();
+            softAssertion
+                    .assertThat(nftAllowanceRepository.findAll(
+                            request, ((EntityIdNumParameter) request.getAccountId()).id()))
+                    .as(testSpec.description())
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:462]: use soft assertion and add the description as the test spec id, so it'll run all specs, and give us enough information to locate which ones have failed.

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 7]
```diff
@@ -188,258 +184,293 @@ private void setupNftAllowances() {
...
     private void assertNftAllowances() {
-        for (var entry : nftAllowanceRequests.entrySet()) {
-            var expectedNftAllowances =
-                    entry.getValue().stream().map(nftAllowances::get).toList();
-
-            var key = entry.getKey();
-            assertThat(nftAllowanceRepository.findAll(key, ((EntityIdNumParameter) key.getAccountId()).id()))
-                    .containsExactlyElementsOf(expectedNftAllowances);
+        var softAssertion = new SoftAssertions();
+        for (int i = 0; i < testSpecs.size(); i++) {
```
Comment (xin-hedera) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:455]: index `i` was added because in one iteration I used `i + 1` as the description to locate the failed test cases...

tend to just leave it as is

**Review by jascks — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
